map/src/test/java/openlr/map/simplemockdb/SimpleMockedLine.java

```diff
@@ -0,0 +1,241 @@
+    }
+
+
+    public static SimpleMockedLine from(openlr.map.simplemockdb.schema.Line xmlLine, Node startNode, Node endNode) {
+        List<LineSegment> lineSegments = generateShape(xmlLine, startNode, endNode);
+        SimpleMockedLine simpleMockedLine = new SimpleMockedLine(xmlLine, startNode, endNode, lineSegments);
+        return simpleMockedLine;
+    }
+
+
+    private SimpleMockedLine(openlr.map.simplemockdb.schema.Line xmlLine, Node startNode, Node endNode,List<LineSegment> segments) {
+        this.startNode = startNode;
+        this.endNode = endNode;
+        this.lineSegments = segments;
+        this.id = xmlLine.getId().longValue();
+        this.restrictions = xmlLine.getRestrictions().stream().map(line -> line.longValue()).collect(Collectors.toList());
+        this.frc = FunctionalRoadClass.getFRCs().get(xmlLine.getFrc());
+        this.fow = FormOfWay.getFOWs().get(xmlLine.getFow());
+        this.hashCode = xmlLine.hashCode();
+    }
+
+    public Node getStartNode() {
+        return this.startNode;
+    }
+
+    public List<Long> getRestrictions() {
+        return this.restrictions;
+    }
+
+    public Node getEndNode() {
+        return this.endNode;
+    }
+
+    public FormOfWay getFOW() {
+
+        return fow;
+    }
+
+    public FunctionalRoadClass getFRC() {
+        return frc;
+    }
+
+    /**
+     * @deprecated
+     */
+    @Deprecated
+    public Point2D.Double getPointAlongLine(int var1) {
+        return null;
+    }
+
+    public GeoCoordinates getGeoCoordinateAlongLine(int distance) throws SimpleMockedException {
+
+        int lengthCovered = 0;
+        for (LineSegment segment : lineSegments) {
+            if (lengthCovered <= distance && lengthCovered + segment.getLength() >= distance) {
+                double offset = distance - lengthCovered;
+                Coordinate point = segment.pointAlong(segment.getLength() / offset);
+                point = segment.project(point);
+                double z = segment.p0.z + 1 * (segment.p1.z - segment.p0.z);
+                point.setOrdinate(2, z);
+                return toWGS84(point);
+            }
+            lengthCovered += segment.getLength();
+        }
+        throw new SimpleMockedException("length is greater than the line length");
+    }
+
+    public int getLineLength() {
+        return (int) this.lineSegments.stream().mapToDouble(segment -> segment.getLength()).sum();
+    }
+
+    public long getID() {
+        return this.id;
+    }
+
+    public Iterator<Line> getPrevLines() {
+        return this.startNode.getIncomingLines();
+    }
+
+    /**
+     * @deprecated use {@link #getShapeCoordinates()}
+     */
+    @Deprecated
+    public java.awt.geom.Path2D.Double getShape() throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
+    }
+
+    public Iterator<Line> getNextLines() {
+        return this.endNode.getOutgoingLines();
+    }
+
+    public boolean equals(Object var1) {
```

Comment:
Add an `@Override` annotation to all `equals()` and `hashCode()`.

Reply:
fixed in 28da152a681e7755c4823adf923ffff52ee07e55